client/state/comments/reducer.js

```diff
@@ -126,7 +127,24 @@ export function totalCommentsCount( state = {}, action ) {
 	return state;
 }
 
+/**
+ * Keeps track of if a comment list request is in flight
+ * @param {object} state  - global redux state
+ * @param {object} action - redux action
+ * @returns {boolean} new redux state
+ */
+export function isCommentListLoading( state = false, action ) {
+	switch ( action.type ) {
+		case COMMENTS_LIST_REQUEST:
+			return true;
+		case COMMENTS_RECEIVE: //we may need a more specific success/error action here
+			return false;
+	}
```

Comment:
@gwwar I'd be strongly opposed to adding a _new_ `SUCCESS` and `_ERROR` message in the data layer. I'm fine with pushing out a pragmatic step, but I really don't want more of those because they hurt us later on.

In these cases where "we just need it" why not just store an extra state value as some kind of enum (string is fine):

```js
import { getData, getError } from 'whatever/util/has/it';

const listState = ( state = null, action ) => {
	if ( COMMENTS_LIST_REQUEST !== action.type ) {
		return state;
	}

	if ( hasData( action ) ) {
		return 'failure';
	}

	if ( hasError( action ) ) {
		return 'loaded';
	}

	return 'loading';
}
```

this is a separate piece of state which we are using to visually indicate something in the UI and it needs no additional action types.